sdk/logical/response.go

```diff
@@ -192,30 +193,127 @@ func RespondWithStatusCode(resp *Response, req *Request, code int) (*Response, e
 	return ret, nil
 }
 
-// HTTPResponseWriter is optionally added to a request object and can be used to
-// write directly to the HTTP response writer.
-type HTTPResponseWriter struct {
+type WrappingResponseWriter interface {
 	http.ResponseWriter
-	written *uint32
+	Wrapped() http.ResponseWriter
+	SetHeaders(h map[string][]*CustomHeader)
+	GetHeaders() map[string][]*CustomHeader
+	StatusCode() int
```

Comment:
Is this actually used?

Reply:
I found the issue in this PR while I was working at the logging requests feature. So, this is not needed now, but it is needed then.

Reply:
You are right, this is not needed. Fixing it in both PRs. Thanks!